commit 493890e75d98810a3470b4aae23be628ee5e9667
Author: Pierre Ossman <drzeus@drzeus.cx>
Date:   Sun Oct 26 12:37:25 2008 +0100

    mmc: increase SD write timeout for crappy cards
    
    It seems that some cards are slightly out of spec and occasionally
    will not be able to complete a write in the alloted 250 ms [1].
    Incease the timeout slightly to allow even these cards to function
    properly.
    
    [1] http://lkml.org/lkml/2008/9/23/390
    
    Signed-off-by: Pierre Ossman <drzeus@drzeus.cx>

diff --git a/drivers/mmc/core/core.c b/drivers/mmc/core/core.c
index 044d84e..f7284b9 100644
--- a/drivers/mmc/core/core.c
+++ b/drivers/mmc/core/core.c
@@ -280,7 +280,11 @@ void mmc_set_data_timeout(struct mmc_data *data, const struct mmc_card *card)
 			(card->host->ios.clock / 1000);
 
 		if (data->flags & MMC_DATA_WRITE)
-			limit_us = 250000;
+			/*
+			 * The limit is really 250 ms, but that is
+			 * insufficient for some crappy cards.
+			 */
+			limit_us = 300000;
 		else
 			limit_us = 100000;
 
